mypy/main.py

```diff
@@ -31,6 +33,40 @@ def __init__(self) -> None:
         self.dirty_stubs = False
 
 
+class MyPyCommand(setuptools.Command):
+    """The :class:`MyPyCommand` class is used by setuptools to perform
+    type checks on registered modules.
+    """
+
+    description = "Validate PEP 0484 Types with mypy"
+    user_options = [
+        ('mypy-args=', None, 'Arguments to pass through to mypy')
+    ]  # type: List[tuple]
+
+    def initialize_options(self):
+        self.mypy_args = ''
+
+    def finalize_options(self):
+        pass
+
+    def run(self):
+        try:
+            mypy_options = []
+            for package in self.distribution.packages:
+                mypy_options.append('--package')
+                mypy_options.append(package)
+            mypy_options.extend(self.mypy_args.split())
```

Comment:
I always hate to see command-line arguments passed through an extra layer of parsing -- it usually means they end up broken in some cases, and this isn't an exception. This approach means that we can't pass through an argument that contains whitespace.

I'm not sure exactly what API we have available to us with `user_options` on `setuptools.Command`. Can we get this as a list of strings instead of a string that we have to split? The classic way to do this kind of pass-through argument, like you see with compilers and linkers, is for the list of arguments to be passed in as a list of arguments -- like `--mypy-arg=foo --mypy-arg=bar` to mean adding arguments `foo` and `bar` to the mypy command line.